src/reanimated2/animation/spring.ts

```diff
@@ -144,13 +293,42 @@ export function withSpring(
       previousAnimation: SpringAnimation
     ): void {
       animation.current = value;
+      animation.startValue = value;
+
+      let newMass = config.mass;
+
+      if (userConfig?.duration) {
+        let x0 = Number(animation.toValue) - value;
```

Comment:
Why we need `Number()` here? We don't do this with other animation types so I'd assume this isn't necessary here and in other places in this file

Reply:
Because we have 
```tsx

export interface AnimationObject {
// ...  
toValue: AnimatableValue;
// ...
}
```
And AnimatableValue is may be a string or array:
```tsx
export type AnimatableValue = number | string | Array<number>;
```